### DISCUSSION TITLE
How to make `eslint-plugin` work when `unocss config file` is not in the project root ?

### DISCUSSION THREAD

**Discussion by yuyinws** [unocss/unocss#2602] [Q&A]
In some projects, frontend and backend code are in the same folder. Such as this:
<img width="198" alt="image" src="https://github.com/unocss/unocss/assets/38490578/b41305f1-c156-4be9-9542-d5249ef97cb7">

The vscode extension provides `unocss.root` to help the extension find the correct config file. But the `eslint-plugin` can not get the config file rightly. It will throw an error:

<img width="1103" alt="image" src="https://github.com/unocss/unocss/assets/38490578/e74891bf-53f6-4221-b476-59c7c310bab2">

Is there a solution to this problem?

**Comment by kilisei** [comment]
A few years later I have the same issue I have looked at the code, and it should be possible, but I cannot get it to work see [code](https://github.com/unocss/unocss/blame/main/packages-integrations/eslint-plugin/src/worker.ts#L99-L104)

But I'm done looking for a solution, maybe it's possible now, and I'm missing it.

It should be something like this, I think:
```typescript
export const makeUnocss = (configPath: string) => {
  configPath = resolve(configPath)
  return {
    name: 'arst/unocss',
    plugins: {
      unocss: pluginUnocss
    },
    rules: pluginUnocss.rules
    settings: {
      configPath, <--- maybe?
    },
  }
}
```
